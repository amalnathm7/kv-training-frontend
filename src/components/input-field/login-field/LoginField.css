.login-field {
    margin-top: 40px;
    align-items: center;
    justify-content: center;
    height: 50px;
    width: 100%;
    border-radius: 5px;
    caret-color: #03AEEE;
    border: 2px solid lightgray;
    text-indent: 10px;
    outline: 0;
}

.login-floating-label-group {
    display: flex;
    flex-direction: column;
    position: relative;
}

.login-floating-label {
    font-size: 14px;
    color: #555555;
    font-weight: bold;
    position: absolute;
    pointer-events: none;
    top: 55px;
    left: 12px;
    transition: all 0.1s ease;
}

.login-field:focus~.login-floating-label,
.login-field:not(:focus):valid~.login-floating-label {
    top: 20px;
    bottom: 0px;
    left: 0px;
    font-size: 11px;
    opacity: 1;
    color: #03AEEE;
}

.login-error {
    margin-top: 10px;
    text-indent: 10px;
    font-size: 14px;
    color: red;
}