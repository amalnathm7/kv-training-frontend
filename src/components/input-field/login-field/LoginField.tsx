import React from 'react';
import './LoginField.css';

export type LoginFieldPropsType = {
  value: string;
  onChange: (event) => void;
  label: string;
  type: 'text' | 'password';
  showError: boolean;
};

const LoginField: React.FC<LoginFieldPropsType> = (props) => {
  return (
    <div className='login-floating-label-group'>
      <input
        className='login-field'
        onChange={props.onChange}
        type={props.type}
        value={props.value}
        required
      ></input>
      <label className='login-floating-label'>{props.label}</label>
      {props.showError && <label className='login-error'>Enter {props.label.toLowerCase()}</label>}
    </div>
  );
};

export default LoginField;
