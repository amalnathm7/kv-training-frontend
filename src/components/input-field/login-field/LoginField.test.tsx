import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import LoginField, { LoginFieldPropsType } from './LoginField';

describe('LoginField Props Test', () => {
  test('If LoginField rendered correctly for username', () => {
    const props: LoginFieldPropsType = {
      value: 'LoginField-value',
      onChange: () => {},
      label: 'LoginField-label',
      showError: false,
      type: 'text'
    };

    const element = render(<LoginField {...props} />);

    expect(element).toMatchSnapshot();
  });

  test('If LoginField rendered correctly for password', () => {
    const props: LoginFieldPropsType = {
      value: 'LoginField-value',
      onChange: () => {},
      label: 'LoginField-label',
      showError: false,
      type: 'password'
    };

    const element = render(<LoginField {...props} />);

    expect(element).toMatchSnapshot();
  });

  test('If LoginField rendered correctly for username with error', () => {
    const props: LoginFieldPropsType = {
      value: 'LoginField-value',
      onChange: () => {},
      label: 'LoginField-label',
      showError: true,
      type: 'text'
    };

    const element = render(<LoginField {...props} />);

    expect(element).toMatchSnapshot();
  });

  test('If LoginField rendered correctly for password with error', () => {
    const props: LoginFieldPropsType = {
      value: 'LoginField-value',
      onChange: () => {},
      label: 'LoginField-label',
      showError: true,
      type: 'password'
    };

    const element = render(<LoginField {...props} />);

    expect(element).toMatchSnapshot();
  });
});
