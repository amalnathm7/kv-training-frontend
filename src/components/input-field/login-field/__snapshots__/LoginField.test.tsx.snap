// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginField Props Test If LoginField rendered correctly for password 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-floating-label-group"
      >
        <input
          class="login-field"
          required=""
          type="password"
          value="LoginField-value"
        />
        <label
          class="login-floating-label"
        >
          LoginField-label
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-floating-label-group"
    >
      <input
        class="login-field"
        required=""
        type="password"
        value="LoginField-value"
      />
      <label
        class="login-floating-label"
      >
        LoginField-label
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LoginField Props Test If LoginField rendered correctly for password with error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-floating-label-group"
      >
        <input
          class="login-field"
          required=""
          type="password"
          value="LoginField-value"
        />
        <label
          class="login-floating-label"
        >
          LoginField-label
        </label>
        <label
          class="login-error"
        >
          Enter 
          loginfield-label
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-floating-label-group"
    >
      <input
        class="login-field"
        required=""
        type="password"
        value="LoginField-value"
      />
      <label
        class="login-floating-label"
      >
        LoginField-label
      </label>
      <label
        class="login-error"
      >
        Enter 
        loginfield-label
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LoginField Props Test If LoginField rendered correctly for username 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-floating-label-group"
      >
        <input
          class="login-field"
          required=""
          type="text"
          value="LoginField-value"
        />
        <label
          class="login-floating-label"
        >
          LoginField-label
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-floating-label-group"
    >
      <input
        class="login-field"
        required=""
        type="text"
        value="LoginField-value"
      />
      <label
        class="login-floating-label"
      >
        LoginField-label
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LoginField Props Test If LoginField rendered correctly for username with error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-floating-label-group"
      >
        <input
          class="login-field"
          required=""
          type="text"
          value="LoginField-value"
        />
        <label
          class="login-floating-label"
        >
          LoginField-label
        </label>
        <label
          class="login-error"
        >
          Enter 
          loginfield-label
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-floating-label-group"
    >
      <input
        class="login-field"
        required=""
        type="text"
        value="LoginField-value"
      />
      <label
        class="login-floating-label"
      >
        LoginField-label
      </label>
      <label
        class="login-error"
      >
        Enter 
        loginfield-label
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
